hepcrawl/crawler2hep.py

```diff
@@ -0,0 +1,178 @@
+            'collaboration',
+            collaboration=collaboration.get('value')
+        )
+
+    for imprint in crawler_record.get('imprints', []):
+        builder.add_if_non_empty(
+            'imprint',
+            imprint_date=imprint.get('date')
+        )
+
+    for copyright in crawler_record.get('copyright', []):
+        builder.add_if_non_empty(
+            'copyright',
+            holder=copyright.get('holder'),
+            material=copyright.get('material'),
+            statement=copyright.get('statement')
+        )
+
+    builder.add_if_non_empty(
+        'preprint_date',
+        preprint_date=crawler_record.get('preprint_date')
+    )
+
+    builder.add_if_non_empty(
+        'acquisition_source',
+        method='hepcrawl',
+        date=crawler_record.get('date'),
+        source=crawler_record.get('source'),
+        submission_number=crawler_record.get('submission_number')
+    )
+
+    try:
+        builder.add_if_non_empty(
+            'number_of_pages',
+            number_of_pages=int(crawler_record.get('page_nr'))
+        )
+    except (TypeError, ValueError):
+        pass
+
+    publication_types = [
+        'introductory',
+        'lectures',
+        'review',
+    ]
+
+    special_collections = [
+        'cdf-internal-note',
+        'cdf-note',
+        'cds',
+        'd0-internal-note',
+        'd0-preliminary-note',
+        'h1-internal-note',
+        'h1-preliminary-note',
+        'halhidden',
+        'hephidden',
+        'hermes-internal-note',
+        'larsoft-internal-note',
+        'larsoft-note',
+        'zeus-internal-note',
+        'zeus-preliminary-note',
+    ]
+
+    document_types = [
+        'book',
+        'note',
+        'report',
+        'proceedings',
+        'thesis',
+    ]
+
+    for collection in crawler_record.get('collections', []):
+        collection = collection['primary'].strip().lower()
+
+        if collection == 'arxiv':
+            continue  # ignored
+        elif collection == 'citeable':
+            builder.set_citeable(True)
+        elif collection == 'core':
+            builder.set_core(True)
+        elif collection == 'noncore':
+            builder.set_core(False)
+        elif collection == 'published':
+            builder.set_refereed(True)
+        elif collection == 'withdrawn':
+            builder.set_withdrawn(True)
+        elif collection in publication_types:
+            builder.add_publication_type(collection)
+        elif collection in special_collections:
+            builder.add_special_collections(collection)
+        elif collection == 'bookchapter':
+            builder.add_document_type('book chapter')
+        elif collection == 'conferencepaper':
+            builder.add_document_type('conference paper')
+        elif collection in document_types:
+            builder.add_document_type(collection)
+
+    if builder.obj.get('document_type') is None:
+        builder.add_document_type('article')
+
```

Comment:
@kaplun send me this lines of code to mange the collections. Anyway I do not like a lot this way to mange the `collections` field because we can manage this also in the spiders without writing these 70 lines of code with a lot of BL inside. In fact the first idea of @kaplun was refactoring this directly in the spiders.
Anyway I am not able to do this refactor in this sprint. Should I open a issue in order to be sure that this thing is going to be refactor in the future? @david-caro